hubblestack/extmods/grains/cloud_details.py

```diff
@@ -35,8 +35,13 @@ def _get_aws_details():
     proxies = {'http': None}
 
     try:
+        ttl_header = {'X-aws-ec2-metadata-token-ttl-seconds': '300'}
+        token_url = 'http://169.254.169.254/latest/api/token'
+        token_request = requests.put(token_url, headers=ttl_header)
+        token = token_request.text
+        aws_token_header = {'X-aws-ec2-metadata-token': token}
```

Comment:
This is new feature released by AWS and although I trust this will almost always work like current metadata APIs but we should still handle scenarios where `token` might be empty due to any reason (why I'm not sue of yet). Also, if token is empty then do we need to switch to older API which works without authentication headers ?

Reply:
I checked this. If the IMDSv1 is enabled then both a valid token and ''(blank) token works. If the IMDSv1 is disabled, then only a valid token will work. So in a way, AWS has put a fallback for empty token value.